README.md

```diff
@@ -72,13 +72,21 @@ filed named `.gitkeep`. Finally, create individual locks by making an empty file
   retrying to acquire a lock or release a lock. The default is 10 seconds.
   Valid values: `60s`, `90m`, `1h`.
 
+* `diff_filter`: *Optional.* If specified, changes the behavior of check. This is an advanced feature of git, refer
```

Comment:
In the past we've found it's better to expose high-level configuration for things like this as it gives us flexibility in how things are implemented. It also makes the UX a bit easier to discover. It's a bit more work up-front but pays off by making it easier to continue supporting the same functionality while refactoring the implementation.

Would you mind having higher level params for this? Seems like it could just translate under the hood to git `diff-filters`. Hopefully it's not much maintenance.

Reply:
Thanks for replying, will do the change later this week. Also appreciated the context you gathered!